server/rest/instance.py

```diff
@@ -120,6 +123,32 @@ def listInstances(self, userId, taleId, text, limit, offset, sort, params):
     def getInstance(self, instance, params):
         return instance
 
+    @access.user
+    @autoDescribeRoute(
+        Description('Updates and restarts an existing instance.')
+        .modelParam('id', model='instance', plugin='wholetale', level=AccessType.WRITE)
+        .errorResponse('ID was invalid.')
+        .errorResponse('Write access was denied for the instance.', 403)
+    )
+    def updateInstance(self, instance):
+        currentUser = self.getCurrentUser()
+        
+        taleId = instance['taleId']
+        if taleId:
```

Comment:
I don't think it's possible to have an instance without a valid `taleId`. Even so, I'd rather raise an error here, rather than pretend everything is fine and return 200.

Reply:
I have added some error handling in the `else` below... do you think that would be sufficient?

Reply:
This condition only checks if `taleId` is not None or empty string, so the execution will still proceed with an invalid id. It will fail in the next line (the one that actually tries to load a model) with a `ValidationException`

```
>>> Tale().load('blah', exc=True)
ValidationException: Invalid ObjectId: blah
```
worth noting that it also throws `ValidationException` for `None` or `''`:
```
>>> Tale().load(None, exc=True)
ValidationException: Attempt to load null ObjectId: None
```
I'd suggest to wrap `.load()` for both Image and Tale with `try: except ValidationError:`,  instead of `if: else: raise`

Reply:
Ahhh, I see what you're saying... I guess I shouldn't assume that any ValidationException thrown would just bubble up to the caller?

Reply:
Actually it bubbles up! I didn't know that. I injected something silly to one of the rest endpoints and I get:

```
$ curl -X GET --header 'Accept: application/json' --header 'Girder-Token: ...' 'https://girder.local.wholetale.org/api/v1/wholetale' -s | jq
{
  "message": "Attempt to load null ObjectId: None",
  "type": "validation"
}
```
No need to even `try/except` it